plugin/kubernetes/reverse.go

```diff
@@ -14,7 +14,11 @@ func (k *Kubernetes) Reverse(state request.Request, exact bool, opt plugin.Optio
 
 	ip := dnsutil.ExtractAddressFromReverse(state.Name())
 	if ip == "" {
-		return nil, nil
+		if dnsutil.IsReverse(state.Name()) {
```

Comment:
This is to shortcut and not perform the `k.Records` lookup? Most likely shouldn't care, and just do the `Records` lookup

Reply:
If the zone is a reverse zone, then we should never do a `k.Records` lookup.
If we did, we would answer things like ...
`s1.ns.svc.in-addr.arpa.` has `PTR`? => `NOERROR`

I may be able to move the reverse check into `k.Records`, that way it's only in one place.

Reply:
> I may be able to move the reverse check into k.Records, that way it's only in one place.

Nevermind - that would end up being uglier... requiring checking the query type again in Records.

Reply:
If the zone is a reverse zone, then we should never do a k.Records lookup.
If we did, we would answer things like ...
s1.ns.svc.in-addr.arpa. has PTR? => NOERROR

Reply:
Still think IsReverse is not needed, ip is empty because we didn't find a reverse so your duplicating that check here.

Reply:
`IsReverse` returns true if name is in a reverse zone.
`ExtractAddressFromReverse` returns an IP if the name is in a reverse zone _and_ contains a valid IP.

So `ExtractAddressFromReverse('s1.ns.svc.in-addr.arpa.')` is ""  because does not contain a valid IP.

While `IsReverse('s1.ns.svc.in-addr.arpa.')` is true because it's in `in-addr.arpa.`

It differentiates these two cases:

- `s1.ns.svc.in-addr.arpa.` has a `PTR`? ...  which should return `NXDOMAIN`
- `s1.ns.svc.cluster.local.` has a `PTR`? ... which should return `NOERROR` (if `s1.ns.svc.cluster.local.` has an `A` or `AAAA` record)

Reply:
But as we are in the Reverse method don't we already know this?

Reply:
Removing the check breaks test...

```
        helpers.go:118: rcode is "NOERROR", expected "NXDOMAIN"
        helpers.go:301: ;; opcode: QUERY, status: NOERROR, id: 4937
                ;; flags: qr aa rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 0
                
                ;; QUESTION SECTION:
                ;svc1.testns.svc.0.10.in-addr.arpa.     IN       A
                
                ;; AUTHORITY SECTION:
                0.10.in-addr.arpa.      300     IN      SOA     ns.dns.0.10.in-addr.arpa. hostmaster.0.10.in-addr.arpa. 0 7200 1800 86400 30

```

Reply:
So doing some awful printf debugging, the actual msg.Service that comes back here is:
~~~
** REVERSE **
reverse
1
/coredns/arpa/in-addr/10/0/svc/testns/svc1
192.168.1.100
~~~
which indicates this should not be solved here.

Reply:
where should it be solved.

Reply:
in k.Records()?